lib/swoosh/adapters/mailgun.ex

```diff
@@ -66,8 +66,20 @@ defmodule Swoosh.Adapters.Mailgun do
     |> prepare_cc(email)
     |> prepare_bcc(email)
     |> prepare_reply_to(email)
+    |> prepare_custom_vars(email)
   end
 
+  # example custom_vars
+  # 
+  # %{"my_var" => Poison.encode!(%{"my_message_id": 123}), 
+  #   "my_other_var" => Poison.encode!(%{"my_other_id": 1, "stuff": 2})}
+  defp prepare_custom_vars(body, %Email{provider_options: custom_vars}) do
+    custom_vars 
+    |> Enum.reduce(body, fn({k, v}, body_acc) -> Map.put(body_acc, "v:#{k}", v) end)
```

Comment:
@stevedomin I'm thinking you mean we should accept inputs like %{"my_var" => %{"my_message_id": 123}} and do JSON encoding inside our function? I agree that could be better since as far as I know only json values are accepted.